File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_table_sizes.c
#include "ses_defines.h"
#include "ses_globals.h"
#include "ses_internals.h"
#include "ses_externs.h"

#include "stdio.h"
#include <stdlib.h>

#include <stdio.h>


//#define DEBUG_WRAP

#ifdef LC_UNDER
ses_boolean ses_get_table_sizes_(ses_file_handle* pt_handle, ses_material_id* mid, ses_table_id_reference the_buffer, long* size) {
#else

#ifdef UC_UNDER
ses_boolean SES_GET_TABLE_SIZES_(ses_file_handle* pt_handle, ses_material_id* mid, ses_table_id_reference the_buffer, long* size) {
#else

#ifdef LC_NOUNDER
ses_boolean ses_get_table_sizes(ses_file_handle* pt_handle, ses_material_id* mid, ses_table_id_reference the_buffer, long* size) {
#else

#ifdef UC_NOUNDER
ses_boolean SES_GET_TABLE_SIZES(ses_file_handle* pt_handle, ses_material_id* mid, ses_table_id_reference the_buffer, long* size) {
#endif

#endif
#endif
#endif

  ses_boolean return_value = SES_TRUE;

  ses_file_handle the_handle = *pt_handle;
  ses_material_id my_mid = *mid;

  long return_size = 0;

#ifdef DEBUG_WRAP
  printf("ses_get_table_sizes.c:  mid is %d\n", my_mid);
#endif

  ses_table_id_reference my_buffer = ses_get_table_sizes(the_handle, my_mid, &return_size);

#ifdef DEBUG_WRAP
  printf("ses_get_table_sizes.c:  return_size is %ld\n", return_size);
  int i = 0;
  for (i = 0; i < return_size; i++) {
	printf("table size[%d] is %d\n", i, my_buffer[i]);
  }
#endif
  int j = 0;
  for (j = 0; j < return_size; j++) {
	the_buffer[j] = my_buffer[j];
  }

  *size = return_size;
#ifdef DEBUG_WRAP
  printf("ses_get_table_sizes.c:  size is %ld return_value is %d\n", *size, return_value);
#endif

  return return_value;

}

 

File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_table_sizes.f90

#undef DEBUG_WRAP

function ses_get_table_sizes_f90(the_handle, mid, the_buffer, the_size)

   implicit none

   SES_FILE_HANDLE, INTENT(IN) :: the_handle
   SES_MATERIAL_ID, INTENT(IN) :: mid
   SES_TABLE_SIZES_REFERENCE, dimension(:)   :: the_buffer
   LONG, INTENT(OUT)           :: the_size

   SES_BOOLEAN :: ses_get_table_sizes
   SES_BOOLEAN :: return_value, ses_get_table_sizes_f90
   SES_BOOLEAN :: didit_getit

   LONG_POINTER :: pt_size
   LONG, target :: my_size

   integer :: i

   SES_TABLE_SIZES_REFERENCE, DIMENSION(:) :: my_buffer
   LONG, DIMENSION(1000), target :: buffer1

   my_buffer=>buffer1

   pt_size=>my_size



#ifdef DEBUG_WRAP
   print *, "ses_get_table_sizes.f90:  calling ses_get_table_sizes"
   print *, "ses_get_table_sizes.f90:  mid is ", mid
#endif

   return_value = ses_get_table_sizes(the_handle, mid, my_buffer, pt_size)
#ifdef DEBUG_WRAP
   print *, "ses_get_table_sizes.f90:  got past function call"
#endif

   the_size = my_size
#ifdef DEBUG_WRAP
   do i = 1, the_size
	print *, "ses_get_table_sizes.f90: value in index ", i, " is ", buffer1(i)
   enddo
#endif


   do i = 1, the_size
	the_buffer(i) = buffer1(i)
   enddo

   ses_get_table_sizes_f90 = return_value

 
end function ses_get_table_sizes_f90


